airmozilla/main/static/main/js/eventstatus.js

```diff
@@ -0,0 +1,19 @@
+(function () {
+    var currentEventStatus = $('.islive').data('event-status');
+    var eventId = $('.islive').data('event-id');
+
+    if (currentEventStatus=='scheduled' || 'pending') {
+        setInterval(function () {
+            refreshIfStatusChanges();
+        }, 30000);
+    }
+
+    function refreshIfStatusChanges() {
+        var url = '/eventstatus/' + eventId;
+        $.get(url, function (data) {
+            if (data.event_status != currentEventStatus) {
+                location.reload();
+            }
+        });
+    }
```

Comment:
You're definitely on the right track and I know the bug didn't mention this but I have another idea. 
Basically, why not apply this business logic. 
```
var url = location.pathname + '/status/';
var eventStatus = null;
var interval = setInterval(function() {
    $.get(url).
    .then(function(data) {
        if (eventStatus !== null && eventStatus !== data.status) {
            location.reload();
        }
        eventStatus = data.data;
    })
    .fail(function() {
        console.error('Failed to fetch event status. Stops looking.');
        clearInterval(interval);
    });
}, 10 * 1000);
```
That's untested and rushed together code but the idea is that after 10 seconds it makes the first load and stores what the current status is and does nothing else. 
Then 10 seconds later it checks what the current status is and whatever is, if it's different from the **first status** that's reason enough to reload. 

This way the code works for all sorts of transitions. From being `scheduled` to `processing` and from `processing` to `scheduled`. Or from `scheduled` to `removed`. And your code won't need to get itself involved at all with the names of statuses. It just looks at differences/changes. Any differences/changes.